src/context/internal.rs

```diff
@@ -122,6 +122,8 @@ pub trait ContextInternal<'a>: Sized {
     fn try_catch_internal<'b: 'a, F>(&mut self, f: F) -> Result<Handle<'a, JsValue>, Handle<'a, JsValue>>
         where F: FnOnce(&mut Self) -> JsResult<'b, JsValue>
     {
+        // A closure does not have a guaranteed layout, so we need to box it in order to pass
+        // a pointer to it across the boundary into C++.
```

Comment:
I don't think I understand this. Isn't the pointer carried through back to Rust where the layout would be known? The static `fn` is generic in the closure.

Pretty old. I was having trouble finding more recent examples.
http://aatch.github.io/blog/2015/01/17/unboxed-closures-and-ffi-callbacks/

As you pointed out elsewhere, try/catch is already slow so this is probably an unnecessary micro optimization. Especially since it's most likely best practice to make `try_catch` closures as small as possible since handling non-`Throw` error types would be awkward.

Reply:
I believe the point of unboxed closures is that `Fn`/`FnMut`/`FnOnce` closures are not guaranteed to be boxed already, so they might be an inline struct of arbitrary size. Which implies that you can't assume they're a pointer.